c10/util/typeid.h

```diff
@@ -442,29 +458,115 @@ class C10_API TypeMeta final {
 #endif
   }
 
- private:
-  const detail::TypeMetaData* data_;
+  /**
+  * convert ScalarType enum values to TypeMeta handles
+  */
+  static inline caffe2::TypeMeta fromScalarType(ScalarType scalar_type) {
+    const size_t index = static_cast<uint16_t>(scalar_type);
+    if (C10_LIKELY(index < NumScalarTypes)) {
```

Comment:
I think you should only do this test in debug mode. Under ordinary circumstances, the `ScalarType` should be enforced, by C++ type system, to be in bounds. I agree it could be corrupted, but I don't think it's worth testing here.

Reply:
Makes sense to me - I basically transcribed the original here on the premise that it was testing a condition that arose in practice, but I agree, it'd be hard to trigger it